trussworks/react-uswds
# !build: Allow direct imports and component normalization

# Summary
This PR makes project-wide changes to allow importing from files directly in addition to the index. Build configuration changes were made to ensure the package passes tests from https://arethetypeswrong.github.io/. All components were modified for one of the following reasons:

* folder normalization (camelcase folder names for folders directly containing a component file only, otherwise lowercase)
* missing exports added
* component props types normalized (BaseXProps and XProps)
* sibling components defined in the same file were moved to dedicated folder/file
* utility functions moved to a `utils` module
* constants moved to a `constants` module
* fixtures moved to a `fixtures` module
* base functional component and forwardRef outputs separated into their own files (X and XForwardRef)
* component modules only have default export

Some areas where we could use one of the react-uswds components instead of a jsx element (ex: Button, Grid) were updated to use those components instead.

## Related Issues or PRs

N/A

## How To Test

* Add the jpandersen87/#tree-shaking branch as your `@trussworks/react-uswds` dependency instead of from npm for your project
* Verify your project continues to work (except for cases where you directly imported types from files that have had their paths changed from normalization)
* Verify you can directly import from files instead of from index
